build_tools/cmake/build_and_test_tsan.sh

```diff
@@ -74,4 +74,11 @@ export IREE_CUDA_DISABLE=1
 # Honor the "notsan" label on tests.
 export IREE_EXTRA_COMMA_SEPARATED_CTEST_LABELS_TO_EXCLUDE=notsan
 
+# Run all tests, once
+"${SCRIPT_DIR}/ctest_all.sh" "${BUILD_DIR}"
+
+# Re-run many times certain tests that are cheap and prone to nondeterministic
+# failure (typically, IREE runtime tests exercising multi-threading features).
+export IREE_CTEST_TESTS_REGEX="(^iree/base/|^iree/task/)"
+export IREE_CTEST_REPEAT_UNTIL_FAIL_COUNT=32
 "${SCRIPT_DIR}/ctest_all.sh" "${BUILD_DIR}"
```

Comment:
sample test time increase: https://github.com/iree-org/iree/runs/7903611071?check_suite_focus=true
```
# all tests
Total Test time (real) = 107.91 sec

# re-run tests
Total Test time (real) =  35.36 sec
```

(not too much of an increase, fine by me)

Reply:
That's not nothing, but I guess it's ok